### PR TITLE
Azure Functions

### PR DESCRIPTION
**Pull request by jelena-music**

### CODE REVIEW

File: Serverless/Azure Functions.md [hunk 1]
```diff
@@ -1,2 +1,156 @@
...
+		    return response;
+		}
+
+
+
+
+#### Function app files 
+Default function app consists of :
+* Function.cs class - Contains function method with defined trigger and input output bindings. ``[FunctionName]`` attribute marks the method as a function entry point. Name must be unique within a project.
+* local.settings.json - Contains all app settings, connectionstrings and settings needed in function development. By default, it contains  ``AzureWebJobsStorage`` key with a value set to ``UseDevelopmentStorage=true``, which is the connection string for the local Azure Storage Emulator.
```
Comment (vperko) [inline Serverless/Azure Functions.md:123]: Let's emphasize that this file is ignored when function is deployed.

File: Serverless/Azure Functions.md [hunk 2]
```diff
@@ -1,2 +1,151 @@
...
+Functions support several different programming languages such as C# and JavaScript, F#, Java, Python, and PowerShell.
+
+In general, Azure Functions are a serverless solution that provide:
+* Writing less code - Allows developer to focus on the business logic implementation and utilize the ready-to-use infrastructure features. 
+* Maintain less infrastructure - Cloud infrastructure providers manage servers, security and scaling. Functions are scaled automatically.
+* Save on costs - No need to buy resources in advance - pay as you go.
+
+
+There are three basic hosting plans available for Azure Functions: 
+* Consumption plan - Functions scale automatically and billing is calculated only for compute resources used i.e. when functions are running. Cold start is startup latency, so first trigger execution after function was idle takes logger time.
```
Comment (nikolic-m) [inline Serverless/Azure Functions.md:11]: Please check if you have meant "longer time" at the end of the sentence.

File: Serverless/Azure Functions.md [hunk 3]
```diff
@@ -1,2 +1,151 @@
...
+               {
+		   "retry":{
+		      "strategy":"fixedDelay",
+		      "maxRetryCount":2,
+		      "delayInterval":"00:00:03"
+		   }
+		}
+```
+
+Note:  Some of the triggers come with default retry logic. Queue triggered functions retry 5 times berfore sending message to poison queue, while the timer trigger function doesn't retry at all. When a time trigger function fails, it isn't called again until the next time on the schedule.
```
Comment (nikolic-m) [inline Serverless/Azure Functions.md:44]: Please correct the typo "berfore".

**Review by nikolic-m — APPROVED** [review]
The content looks good. Please check and correct typos before proceeding.

File: Serverless/Azure Functions.md [hunk 4]
```diff
@@ -1,2 +1,151 @@
-## Azure Functions
+Azure Functions is a Function-as-a-Service (“FaaS”), i.e. a platform for running configurable blocks of code that run in response to some event.
+Functions support several different programming languages such as C# and JavaScript, F#, Java, Python, and PowerShell.
+
+In general, Azure Functions are a serverless solution that provide:
+* Writing less code - Allows developer to focus on the business logic implementation and utilize the ready-to-use infrastructure features. 
+* Maintain less infrastructure - Cloud infrastructure providers manage servers, security and scaling. Functions are scaled automatically.
+* Save on costs - No need to buy resources in advance - pay as you go.
```
Comment (rubber-duck) [inline Serverless/Azure Functions.md:7]: Maybe reword the first part to something like "Main benefits of using Azure Functions and serverless solutions are :" because the two bullets don't match wording of "that provide : ".

File: Serverless/Azure Functions.md [hunk 5]
```diff
@@ -1,2 +1,151 @@
...
+Note: If there is already existing App service hosting some API that needs some background job to be executed, good candidate can be Azure Web job.
+To learn more, see [here](https://docs.microsoft.com/en-us/azure/app-service/webjobs-create).
+
+
+### Scaling 
+ A single function app in Consumption plan, scales out to the maximum of 200 instances, while a single instance can process more than one message / request at a time.
+ 
+Timer trigger functions are singleton (trigger only one instance of function).
+
+In the real world scenarios we need to control scaling of instances. Sometimes azure functions deal with databases or other services which may have scaling limits so over-scaling azure functions could put other resources under pressure. Scaling rules can be configured for function depending on trigger type.
```
Comment (rubber-duck) [inline Serverless/Azure Functions.md:24]: In ~~the~~ real world

File: Serverless/Azure Functions.md [hunk 6]
```diff
@@ -1,2 +1,151 @@
...
+
+
+### Scaling 
+ A single function app in Consumption plan, scales out to the maximum of 200 instances, while a single instance can process more than one message / request at a time.
+ 
+Timer trigger functions are singleton (trigger only one instance of function).
+
+In the real world scenarios we need to control scaling of instances. Sometimes azure functions deal with databases or other services which may have scaling limits so over-scaling azure functions could put other resources under pressure. Scaling rules can be configured for function depending on trigger type. 
+ 
+Example: If for some reason, we need one queue message at the time to be processed by queue triggered function than, we need to configure function app scale-out to one function instance. Than the function concurent proccessing should be limited too, and that is done by setting ``batchSize`` to 1 in the host.json. Learn more [here](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-queue-trigger?tabs=csharp#concurrency) and [here](https://docs.microsoft.com/en-us/azure/azure-functions/functions-concurrency).
```
Comment (rubber-duck) [inline Serverless/Azure Functions.md:26]: we needed queue messages to be processed a triggered function one at a time

**Review by rubber-duck — APPROVED** [review]
